src/pages/docs/creators/ray/cluster-yaml.md

```diff
@@ -104,4 +104,88 @@ By default, Ray on Golem uses prepackaged VM images including [relatively fresh]
 However, you can use these properties to override the detection and request a specific image. 
 Supported tags are available on [Golem registry](https://registry.golem.network/explore/golem/ray-on-golem).
 
-Please [let us know on `#Ray on Golem` discord channel)](https://chat.golem.network/) if you need an image with any specific content. We will be happy to help you.
+Please [let us know on the `#Ray on Golem` discord channel)](https://chat.golem.network/) if you need an image with any specific content. We will be happy to help you.
+
+## Budget management properties
+
+### Spending hard limit
+
+Within `provider.parameters` section there is the `budget` property.
+It defines the maximum amount of GLMs paid for the whole cluster operations - from `ray up` until `ray down`.
+
+At the moment, when the spending reaches the limit, Ray on Golem will stop spending, effectively terminating the cluster nodes.
+
+```yaml
+provider:
+  parameters:
+
+    # Maximum amount of GLMs that's going to be spent for the whole cluster
+    budget: 2
+```
+
+### Avoiding too-expensive providers
+
+You can use `provider.parameters.node_config.cost_management` section to define the limits on providers' prices.
+Ray on Golem won't work with providers who exceed any of the following price settings.
+
+#### Maximum provider prices
+
+Golem providers charge in three ways. They charge:
+- the initial price at the start of image deployment,
+- CPU usage price for the total time (in seconds) their CPUs spent computing,
+- duration price for the total time (in seconds) they spent up and running,
+
+So for example, if you rented a 3 CPU node for 15 minutes and utilized all three cores for 10 minutes you will be charged `initial_price + duration_price * 15*60 + cpu_usage_price * 3 * 10*60`.
```

Comment:
Hard to imagine you will use CPUs for 5 minutes less then the machine.

It is more like: YOu rent 3 CPU but average load in this 1 5minutes is 0.8 per CPU (or 3*0.8 = 2.4 CPU usage) giving you cpu_usage_price * 2.4 * 15*60

Reply:
well, it's _not_ very hard to imagine... cpu time is not the wall time but the _actual_ time that a given CPU is being used and the execution may well be memory- or io- extensive...